Tensile/Source/lib/source/ContractionSolution.cpp

```diff
@@ -750,7 +752,61 @@ namespace Tensile
         for(size_t i = 0; i < problem.batchIndices().size(); i++)
             wiZ *= problem.batchSize(i);
 
-        rv.numWorkGroups.x = CeilDivide(wiX * wiY * wiZ, rv.workGroupSize.x);
+        unsigned int gsu = static_cast<int>(sizeMapping.globalSplitU);
+        if(sizeMapping.globalAccumulation == 1)
+            // globalAccumulation = 1 case, ignore globalSplitU and use 1
+            gsu = 1;
+        // wider global load for postGSU
+        // only for compute type = Float
+        bool supportedTypeForVWopt
+            = problem.alphaType() == DataType::Float || problem.alphaType() == DataType::Double;
+        size_t total     = wiX * wiY * wiZ;
+        int    vw        = 1;
+        size_t threshVW2 = cuCount * 256 * 2; // should be more than number of physical threads * vw
+        if(supportedTypeForVWopt && total > threshVW2 && problem.freeSizeA(0) % 2 == 0)
+            vw = 2;
+
+        // parallel reduction width calculation
+        // only for compute type = Float
+        bool supportedTypeForReductionOpt
+            = problem.alphaType() == DataType::Float || problem.alphaType() == DataType::Double;
+        size_t threshReduction
+            = cuCount * 256; // should be less than number of physical threads / reduction
+        const unsigned int maxReductionConst = 4;
+        const unsigned int minGSUperReduction
+            = 32; // Minimum GSU=128 for Reduction=4, GSU=64 for Reduction2
```

Comment:
Lots of magic numbers here make this a bit less clear to read

Reply:
I will replace numbers with some meaningful names.
  256 -> numThreadsPerCU
Anything else I need to update?

Reply:
I wasn't too sure from the comments what the 32 is on line 777

Reply:
But maybe that's more about the comments since they say mingsu=128 or 64?

Reply:
I thought I already named it as const unsigned int minGSUperReduction.
Reduction did not perform well with small GSU.

So, I defined minimum number of GSU to apply reduction optimization.
  Reduction=2 case, min GSU for reduction = 64
  Reduction=4 case, min GSU for reduction = 128
Here, minGSUperReduction defines minimum GSU / Reduction = 32.
Didn't I put enough explanation here?